kuksa-sdk/src/main/kotlin/org/eclipse/kuksa/DataBrokerApiInteraction.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa
+
+import io.grpc.Context
+import io.grpc.ManagedChannel
+import io.grpc.StatusRuntimeException
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.eclipse.kuksa.proto.v1.KuksaValV1
+import org.eclipse.kuksa.proto.v1.Types
+import org.eclipse.kuksa.proto.v1.Types.Field
+import org.eclipse.kuksa.proto.v1.VALGrpc
+import org.eclipse.kuksa.subscription.Subscription
+
+/**
+ * Encapsulates the interactions with the DataBroker using gRPC.
+ */
+internal class DataBrokerApiInteraction(
```

Comment:
Api interaction sounds like a naming which could be used for everything.^^ "DataBrokerChannel" maybe because of the tight coupling to the ManagedChannel but providing addition DataBroker API?

Reply:
Renamed to DataBrokerTransporter